### PR TITLE
[connection-status] Use Websockets

### LINKED ISSUE #1490: alive check is spamming network tab in the dev tools
We could consider using ws ping/pong instead.

### PR DESCRIPTION
**Pull request by svenefftinge**
Changed the connection status service to use websockets
and natural activity. (fixes #1490)

### CODE REVIEW

**Comment by kittaakos** [comment]
I am doing the review now...

Please note, your connection-status tests have failed on the CIs.

File: packages/core/src/browser/connection-status-service.ts [hunk 1]
```diff
@@ -62,92 +50,42 @@ export enum ConnectionState {
...
-    protected connectionState: ConnectionStatusImpl;
+    protected connectionState: ConnectionState = ConnectionState.ONLINE;
     protected timer: number | undefined;
-    protected retryInterval: number;
 
     constructor(
         @inject(ConnectionStatusOptions) @optional() protected readonly options: ConnectionStatusOptions = ConnectionStatusOptions.DEFAULT,
         @inject(ILogger) protected readonly logger: ILogger
     ) {
-        this.statusChangeEmitter = new Emitter<ConnectionStatus>();
```
Comment (kittaakos) [inline packages/core/src/browser/connection-status-service.ts]: Can you please switch to property injection then?

File: packages/core/src/browser/connection-status-service.ts [hunk 2]
```diff
@@ -217,6 +130,42 @@ export class FrontendConnectionStatusService implements ConnectionStatusService,
...
 }
 
+@injectable()
+export class FrontendConnectionStatusService extends AbstractConnectionStatusService {
+
+    private scheduledPing: number | undefined;
+
+    constructor(
+        @inject(WebSocketConnectionProvider) protected readonly wsConnectionProvider: WebSocketConnectionProvider,
+        @inject(PingService) protected readonly pingservice: PingService,
```
Comment (kittaakos) [inline packages/core/src/browser/connection-status-service.ts]: `pingservice` -> `pingService` please.

File: packages/core/src/browser/connection-status-service.ts [hunk 3]
```diff
@@ -158,54 +96,29 @@ export class FrontendConnectionStatusService implements ConnectionStatusService,
...
+        }
+        this.logger.trace(success ? `Connected to the backend.` : `Cannot reach the backend.`);
+        const previousState = this.connectionState;
+        const newState = success ? ConnectionState.ONLINE : ConnectionState.OFFLINE;
+        if (previousState !== newState) {
+            this.connectionState = newState;
+            this.fireStatusChange(newState);
+        }
+        // schedule offline
+        this.timer = this.setTimeout(async () => {
```
Comment (kittaakos) [inline packages/core/src/browser/connection-status-service.ts]: Unnecessary `async`.

File: packages/core/src/browser/connection-status-service.ts [hunk 4]
```diff
@@ -217,6 +130,42 @@ export class FrontendConnectionStatusService implements ConnectionStatusService,
 
 }
 
+@injectable()
+export class FrontendConnectionStatusService extends AbstractConnectionStatusService {
+
+    private scheduledPing: number | undefined;
+
+    constructor(
```
Comment (kittaakos) [inline packages/core/src/browser/connection-status-service.ts]: Please use property injection.

File: packages/core/src/browser/connection-status-service.ts [hunk 5]
```diff
@@ -30,29 +30,17 @@ export interface ConnectionStatusService {
...
      * The actual connection state.
      */
-    readonly currentState: ConnectionStatus;
+    readonly currentState: ConnectionState;
 
     /**
      * Clients can listen on connection status change events.
      */
-    readonly onStatusChange: Event<ConnectionStatus>;
+    readonly onStatusChange: Event<ConnectionState>;
```
Comment (kittaakos) [inline packages/core/src/browser/connection-status-service.ts]: It should be `onStateChange`.

File: packages/core/src/browser/connection-status-service.ts (hunk 1 above)
Comment (svenefftinge) [reply packages/core/src/browser/connection-status-service.ts]: I want to use it without injection (see mock-connection-service.ts)

Comment (kittaakos) [reply packages/core/src/browser/connection-status-service.ts]: I see, then it makes sense. 👍

File: packages/core/src/browser/connection-status-service.ts (hunk 4 above)
Comment (svenefftinge) [reply packages/core/src/browser/connection-status-service.ts]: see above

File: packages/core/src/browser/connection-status-service.ts (hunk 5 above)
Comment (svenefftinge) [reply packages/core/src/browser/connection-status-service.ts]: I changed `state` to `status`

**Comment by kittaakos** [comment]
It works nicely. The tests are passing (locally) too. I have one more question; shouldn't we bind the `PingService` `inSingletonScope`?

**Comment by svenefftinge** [comment]
> shouldn't we bind the PingService inSingletonScope?

There is no reason, it will be created once as it is requested only once. 
And even if there was another client it doesn't matter to have it twice, because it is stateless.

**Comment by kittaakos** [comment]
> client it doesn't matter to have it twice, because it is stateless.

Alright, let's leave as is.

**Pull request merged**